float64 input_x
float64 input_y
---
float64 output_x
float64 output_y
float64 output_z
float64 output_w